test/e2e/assessment_helpers_test.go

```diff
@@ -351,3 +367,83 @@ func testErrorEmpty(err error) bool {
 		return false
 	}
 }
+
+func assessTestCommands(ctx context.Context, client klient.Client, pod *v1.Pod, testCommands []testCommand) (string, error) {
+	var podlist v1.PodList
+	if err := client.Resources(pod.Namespace).List(ctx, &podlist); err != nil {
+		return "Failed to list pod", err
+	}
+	for _, testCommand := range testCommands {
+		var stdout, stderr bytes.Buffer
+		for _, podItem := range podlist.Items {
+			if podItem.ObjectMeta.Name == pod.Name {
```

Comment:
What's expected result if no Pod found with same Name? pass or fail?

Reply:
Given we have return error when failing to create Pod, I think it's OK here. But looks missed `break`.